ngx_lua_sample/nginx.conf

```
@@ -0,0 +1,110 @@
+# vi:ft=nginx:
+events {
+  worker_connections 1024;
+}
+
+error_log stderr;
+
+http {
+  resolver 127.0.0.11 ipv6=off;
+
+  lua_package_path "/usr/local/openresty/lualib/?.lua;/usr/local/openresty/luajit/share/lua/5.1/?.lua;/lua/src/?.lua";
+  lua_package_cpath "/usr/local/openresty/lualib/?.so;/usr/local/openresty/luajit/lib/lua/5.1/?.so;";
+
+
+  init_by_lua_block {
+    -- just forcing the bytecode pre-compilation
+    -- source: https://blog.openresty.com/en/luajit-bytecode/
+    require "resty.http"
+    require "lua-circuit-breaker-ngx-usage"
+  }
+
+  # simultaing server using the circuit breaker lib
+  server {
+    listen 8080;
+
+    location / {
+      content_by_lua_block {
+        local httpc = require("resty.http").new()
+        httpc:set_timeout(300) -- 300ms timeout
+
+        local circuit_breakers = require "lua-circuit-breaker-ngx-usage"
+        local cb = circuit_breakers.get_circuit_breaker("my_cb", "my_group")
+
+        -- Check state of cb. This function returns an error if the state is open or half_open_max_calls_in_window is breached.
+        local _, err_cb = cb:_before()
+        if err_cb then
+            ngx.exit(ngx.HTTP_INTERNAL_SERVER_ERROR)
```

```suggestion
          ngx.status = 500

          local err = {
            "Circuit breaker is open!"
          }
          ngx.print(err)
          ngx.exit(ngx.OK)
          return
```

Should we send a message in the body that the CB is open instead of 500 Error? Status would still be 500.